activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -494,6 +494,21 @@ def initialize_type_map(m) # :nodoc:
             end
           end
 
+          if ActiveRecord::Base.postgresql_interval_type_as_duration
+            m.register_type 'interval' do |*args, sql_type|
+              precision = extract_precision(sql_type)
+              OID::Interval.new(precision: precision)
+            end
+          else
+            ActiveSupport::Deprecation.warn(<<-MSG.squish)
+              Working with PostgreSQL interval datatype as with varchar is deprecated!
+              Starting from 5.1 interval values will be converted to and from ActiveSupport::Duration objects by default.
+              To use new behavior please set ActiveRecord::Base.postgresql_interval_type_as_duration to true in config/application.rb
+              See https://github.com/rails/rails/pull/16919 for details.
```

Comment:
This will print out a deprecation warning for all apps on boot. We should do this as I suggested in my previous comment so it only affects people who are currently using `interval` columns.